Source/dllmain.cpp

```diff
@@ -0,0 +1,114 @@
+#include <fstream>
+#include <comdef.h>
+#include <string>
+#include <sstream>
+#include <thread>
+#include <mutex>   
+#include <condition_variable>
+#include "windows.h"
+#include "TlHelp32.h"
+#pragma warning(disable : 4996)
+
+HHOOK g_hHook = NULL;
+HINSTANCE g_hDll = NULL;
+
+BOOL APIENTRY DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved);
+extern "C" __declspec(dllexport) BOOL SetMsgHook(char* path);    // Exported
+extern "C" __declspec(dllexport) BOOL UnsetMsgHook();   
+extern "C" __declspec(dllexport) BOOL OpenLogFile(char* path);
+LRESULT CALLBACK SysMsgProc(int nCode, WPARAM wParam, LPARAM lParam);
+
+std::ofstream myfile;
+
+extern "C" __declspec(dllexport) HINSTANCE getDllHinstance() {
+	return g_hDll;
+}
+
+void printLog(const std::string& message)
+{
+    if (myfile.is_open()) {
+        myfile << message << std::endl;
+    }
+}
+
+BOOL APIENTRY DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved)
+{
+	switch (fdwReason)
+	{
+	case DLL_PROCESS_ATTACH:
+		g_hDll = hinstDLL;
+		return true;
+    case DLL_PROCESS_DETACH:
+        UnsetMsgHook();
+        return true;
```

Comment:
Looks like different indentation. Should be always 4 spaces (proper settings in VS and `Alt+F8` shortcut) not tabs.